### PR TITLE
Build updates and compliant release pipeline

### PR DESCRIPTION
**Pull request by andschwa**
Sets up a compliant build pipeline to do a new release of this project.

### CODE REVIEW

**Comment by tig** [comment]
I believe other defunct artifacts include:

- `./tools/releaseBuild`
- `./assets`
- 

Also, this part of `./README.md` is not quite accurate and may confuse peeps:

```
## (Deprecated) Microsoft.PowerShell.GraphicalTools Architecture

Due to the quirks of the PowerShell threading implementation, the design of
GUIs in this application are non-standard. The cmdlet invokes an Avalonia
application as a separate process to guarantee the GUI is running on the main
thread. Graphical tools therefore consists of 3 .NET Projects.

- Microsoft.PowerShell.GraphicalTools - Cmdlet implementations
- OutGridView.Gui - Implementation of the Out-GridView window
- OutGridView.Models - Contains data contracts between the GUI & Cmdlet
```

I'd write it as

## Microsoft.PowerShell.GraphicalTools Architecture

`GraphicalTools' consists of 3 .NET Projects:

- `ConsoleGuiTools` - Cmdlet implementation for `Out-ConsoleGridView`
- `OutGridView.Models` - Contains data contracts between the GUI & Cmdlet

_Note:_ Previously, `GraphicalTools` also included the Avalonia-based `Out-GridView` which was implemented in `.\Microsoft.PowerShell.GraphicalTools` and `.\OutGridView.Gui`. These components have been deprecated (see note above).

File: src/Microsoft.PowerShell.GraphicalTools/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Deprecation Notice
+
+This module of the project, despite being the namesake, is currently deprecated due to the
+Avalonia dependency being too far out of date. One day we may update it using another
+library, but for now it is left as just historical code. This repository now only ships
+the "CUI" or Console User Interface modules.
```
[the remark this answers is not in the document]
Comment (andschwa) [reply src/Microsoft.PowerShell.GraphicalTools/README.md:6]: Honestly, yeah, I'd be fine deleting it all too and moving this to the root readme.

[the remark this answers is not in the document]
Comment (andschwa) [reply src/Microsoft.PowerShell.GraphicalTools/README.md]: Deleted!

File: .vsts-ci/azure-pipelines-release.yml [hunk 2]
```diff
@@ -12,158 +9,68 @@ variables:
...
-      - .editorconfig
-      - .gitattributes
-      - .gitignore
-      - /docs/**/*
-      - /CHANGELOG.md
-      - /CONTRIBUTING.md
-      - /README.md
-      - /LICENSE.txt
-      - /CODE_OF_CONDUCT.md
+      - release
```
Comment (adityapatwardhan) [inline .vsts-ci/azure-pipelines-release.yml:14]: do you want this to be release/*

File: .vsts-ci/templates/release-general.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    archiveFile: GraphicalTools.zip
+    replaceExistingArchive: true
+    verbose: true
+
+- checkout: self
+
+- template: assembly-module-compliance.yml@ComplianceRepo
+  parameters:
+    # binskim
+    AnalyzeTarget: $(Pipeline.Workspace)/*.dll
```
Comment (adityapatwardhan) [inline .vsts-ci/templates/release-general.yml:71]: Could you verify if this path actually has the DLLs.

File: .vsts-ci/azure-pipelines-release.yml (hunk 2 above)
Comment (andschwa) [reply .vsts-ci/azure-pipelines-release.yml:14]: I do not, I just create a temporary release branch when I go to make a release, thanks though!

File: .vsts-ci/templates/release-general.yml (hunk 3 above)
Comment (andschwa) [reply .vsts-ci/templates/release-general.yml:71]: Yes, verified! I see:

```
Analyzing 'Microsoft.PowerShell.ConsoleGuiTools.dll'...
```
as well as all the other DLLs.

**Pull request merged**